pkgs/development/libraries/mesa/default.nix

```
@@ -23,12 +23,14 @@
 , libvdpau
 , llvmPackages
 , lm_sensors
+, mesa
```

> It's not bootstrapped, I'm using a native (for the build machine) build of Mesa to provide the intel-clc tool that is needed for cross builds.

How about adding the executable path into [`disallowedReferences`](https://nixos.org/manual/nix/stable/language/advanced-attributes#adv-attr-disallowedRequisites) to make sure that the `outPath`s are free from such references?

To avoid explicitly using the `.__spliced` attribute for such purpose, consider using `mesaNative` instead of `mesa` as a `<pkg>.override` argument, and specify `{ mesaNative = buildPackages.mesa; }` inside `all-packages.nix`.

```suggestion
, mesaNative
```

---

I realized we already have `buildPackages` so we don't actually need any of that - we can just get `buildPackages.mesa` everywhere.

---

> we can just get buildPackages.mesa everywhere.

Wouldn't it make the package harder to `<pkg>.override`?

---

Yes, but that's a Nixpkgs-wide problem.  It doesn't really make sense to work around it one package at a time.